### ISSUE TITLE
`margin` has been changed to `svm_margin` in the latest version of dials

### ISSUE THREAD

**Issue by UnclAlDeveloper** [tidymodels/parsnip#330]
```{r}
library(tidymodels)
#> -- Attaching packages ------------------------------------------------------------------------------------------------------------------- tidymodels 0.1.0 --
#> v broom     0.5.6      v recipes   0.1.12
#> v dials     0.0.7      v rsample   0.0.7 
#> v dplyr     1.0.0      v tibble    3.0.1 
#> v ggplot2   3.3.1      v tune      0.1.0 
#> v infer     0.5.1      v workflows 0.1.1 
#> v parsnip   0.1.1      v yardstick 0.0.6 
#> v purrr     0.3.4
#> -- Conflicts ---------------------------------------------------------------------------------------------------------------------- tidymodels_conflicts() --
#> x purrr::discard() masks scales::discard()
#> x dplyr::filter()  masks stats::filter()
#> x dplyr::lag()     masks stats::lag()
#> x recipes::step()  masks stats::step()

model <- svm_rbf(mode = "regression", cost = tune(), rbf_sigma = tune(), margin = tune()) %>% set_engine("kernlab")

parameters(model)
#> Error: Problem with `mutate()` input `object`.
#> x Error when calling margin(): Error : 'margin' is not an exported object from 'namespace:dials'
#> i Input `object` is `purrr::map(call_info, eval_call_info)`.
Created on 2020-06-15 by the reprex package (v0.3.0)

Session info
devtools::session_info()
#> - Session info ---------------------------------------------------------------
#>  setting  value                       
#>  version  R version 3.6.3 (2020-02-29)
#>  os       Windows Server x64          
#>  system   x86_64, mingw32             
#>  ui       RTerm                       
#>  language (EN)                        
#>  collate  English_United States.1252  
#>  ctype    English_United States.1252  
#>  tz       America/New_York            
#>  date     2020-06-15                  
#> 
#> - Packages -------------------------------------------------------------------
#>  package       * version    date       lib source        
#>  assertthat      0.2.1      2019-03-21 [1] CRAN (R 3.6.1)
#>  backports       1.1.7      2020-05-13 [1] CRAN (R 3.6.3)
#>  base64enc       0.1-3      2015-07-28 [1] CRAN (R 3.6.0)
#>  bayesplot       1.7.2      2020-05-28 [1] CRAN (R 3.6.3)
#>  boot            1.3-24     2019-12-20 [2] CRAN (R 3.6.3)
#>  broom         * 0.5.6      2020-04-20 [1] CRAN (R 3.6.3)
#>  callr           3.4.3      2020-03-28 [1] CRAN (R 3.6.3)
#>  class           7.3-15     2019-01-01 [2] CRAN (R 3.6.3)
#>  cli             2.0.2      2020-02-28 [1] CRAN (R 3.6.3)
#>  codetools       0.2-16     2018-12-24 [2] CRAN (R 3.6.3)
#>  colorspace      1.4-1      2019-03-18 [1] CRAN (R 3.6.1)
#>  colourpicker    1.0        2017-09-27 [1] CRAN (R 3.6.3)
#>  crayon          1.3.4      2017-09-16 [1] CRAN (R 3.6.1)
#>  crosstalk       1.1.0.1    2020-03-13 [1] CRAN (R 3.6.3)
#>  desc            1.2.0      2018-05-01 [1] CRAN (R 3.6.1)
#>  devtools        2.3.0      2020-04-10 [1] CRAN (R 3.6.3)
#>  dials         * 0.0.7      2020-06-10 [1] CRAN (R 3.6.3)
#>  DiceDesign      1.8-1      2019-07-31 [1] CRAN (R 3.6.3)
#>  digest          0.6.25     2020-02-23 [1] CRAN (R 3.6.3)
#>  dplyr         * 1.0.0      2020-05-29 [1] CRAN (R 3.6.3)
#>  DT              0.13       2020-03-23 [1] CRAN (R 3.6.3)
#>  dygraphs        1.1.1.6    2018-07-11 [1] CRAN (R 3.6.3)
#>  ellipsis        0.3.1      2020-05-15 [1] CRAN (R 3.6.3)
#>  evaluate        0.14       2019-05-28 [1] CRAN (R 3.6.0)
#>  fansi           0.4.1      2020-01-08 [1] CRAN (R 3.6.2)
#>  fastmap         1.0.1      2019-10-08 [1] CRAN (R 3.6.1)
#>  foreach         1.5.0      2020-03-30 [1] CRAN (R 3.6.3)
#>  fs              1.4.1      2020-04-04 [1] CRAN (R 3.6.3)
#>  furrr           0.1.0      2018-05-16 [1] CRAN (R 3.6.3)
#>  future          1.17.0     2020-04-18 [1] CRAN (R 3.6.3)
#>  generics        0.0.2      2018-11-29 [1] CRAN (R 3.6.1)
#>  ggplot2       * 3.3.1      2020-05-28 [1] CRAN (R 3.6.3)
#>  ggridges        0.5.2      2020-01-12 [1] CRAN (R 3.6.3)
#>  globals         0.12.5     2019-12-07 [1] CRAN (R 3.6.1)
#>  glue            1.4.1      2020-05-13 [1] CRAN (R 3.6.3)
#>  gower           0.2.1      2019-05-14 [1] CRAN (R 3.6.0)
#>  GPfit           1.0-8      2019-02-08 [1] CRAN (R 3.6.3)
#>  gridExtra       2.3        2017-09-09 [1] CRAN (R 3.6.1)
#>  gtable          0.3.0      2019-03-25 [1] CRAN (R 3.6.1)
#>  gtools          3.8.2      2020-03-31 [1] CRAN (R 3.6.3)
#>  highr           0.8        2019-03-20 [1] CRAN (R 3.6.1)
#>  htmltools       0.4.0      2019-10-04 [1] CRAN (R 3.6.1)
#>  htmlwidgets     1.5.1      2019-10-08 [1] CRAN (R 3.6.1)
#>  httpuv          1.5.4      2020-06-06 [1] CRAN (R 3.6.3)
#>  igraph          1.2.5      2020-03-19 [1] CRAN (R 3.6.3)
#>  infer         * 0.5.1      2019-11-19 [1] CRAN (R 3.6.3)
#>  inline          0.3.15     2018-05-18 [1] CRAN (R 3.6.3)
#>  ipred           0.9-9      2019-04-28 [1] CRAN (R 3.6.0)
#>  iterators       1.0.12     2019-07-26 [1] CRAN (R 3.6.1)
#>  janeaustenr     0.1.5      2017-06-10 [1] CRAN (R 3.6.1)
#>  knitr           1.28       2020-02-06 [1] CRAN (R 3.6.3)
#>  later           1.1.0.1    2020-06-05 [1] CRAN (R 3.6.3)
#>  lattice         0.20-40    2020-02-19 [2] CRAN (R 3.6.3)
#>  lava            1.6.7      2020-03-05 [1] CRAN (R 3.6.3)
#>  lhs             1.0.2      2020-04-13 [1] CRAN (R 3.6.3)
#>  lifecycle       0.2.0      2020-03-06 [1] CRAN (R 3.6.3)
#>  listenv         0.8.0      2019-12-05 [1] CRAN (R 3.6.1)
#>  lme4            1.1-23     2020-04-07 [1] CRAN (R 3.6.3)
#>  loo             2.2.0      2019-12-19 [1] CRAN (R 3.6.3)
#>  lubridate       1.7.9      2020-06-08 [1] CRAN (R 3.6.3)
#>  magrittr        1.5        2014-11-22 [1] CRAN (R 3.6.1)
#>  markdown        1.1        2019-08-07 [1] CRAN (R 3.6.1)
#>  MASS            7.3-51.6   2020-04-26 [1] CRAN (R 3.6.3)
#>  Matrix          1.2-18     2019-11-27 [2] CRAN (R 3.6.3)
#>  matrixStats     0.56.0     2020-03-13 [1] CRAN (R 3.6.3)
#>  memoise         1.1.0      2017-04-21 [1] CRAN (R 3.6.1)
#>  mime            0.9        2020-02-04 [1] CRAN (R 3.6.2)
#>  miniUI          0.1.1.1    2018-05-18 [1] CRAN (R 3.6.3)
#>  minqa           1.2.4      2014-10-09 [1] CRAN (R 3.6.1)
#>  munsell         0.5.0      2018-06-12 [1] CRAN (R 3.6.1)
#>  nlme            3.1-145    2020-03-04 [2] CRAN (R 3.6.3)
#>  nloptr          1.2.2.1    2020-03-11 [1] CRAN (R 3.6.3)
#>  nnet            7.3-13     2020-02-25 [2] CRAN (R 3.6.3)
#>  parsnip       * 0.1.1      2020-05-06 [1] CRAN (R 3.6.3)
#>  pillar          1.4.4      2020-05-05 [1] CRAN (R 3.6.3)
#>  pkgbuild        1.0.8      2020-05-07 [1] CRAN (R 3.6.3)
#>  pkgconfig       2.0.3      2019-09-22 [1] CRAN (R 3.6.1)
#>  pkgload         1.1.0      2020-05-29 [1] CRAN (R 3.6.3)
#>  plyr            1.8.6      2020-03-03 [1] CRAN (R 3.6.3)
#>  prettyunits     1.1.1      2020-01-24 [1] CRAN (R 3.6.2)
#>  pROC            1.16.2     2020-03-19 [1] CRAN (R 3.6.2)
#>  processx        3.4.2      2020-02-09 [1] CRAN (R 3.6.3)
#>  prodlim         2019.11.13 2019-11-17 [1] CRAN (R 3.6.1)
#>  promises        1.1.1      2020-06-09 [1] CRAN (R 3.6.3)
#>  ps              1.3.3      2020-05-08 [1] CRAN (R 3.6.3)
#>  purrr         * 0.3.4      2020-04-17 [1] CRAN (R 3.6.3)
#>  R6              2.4.1      2019-11-12 [1] CRAN (R 3.6.1)
#>  Rcpp            1.0.4.6    2020-04-09 [1] CRAN (R 3.6.3)
#>  RcppParallel    5.0.1      2020-05-06 [1] CRAN (R 3.6.3)
#>  recipes       * 0.1.12     2020-05-01 [1] CRAN (R 3.6.3)
#>  remotes         2.1.1      2020-02-15 [1] CRAN (R 3.6.3)
#>  reshape2        1.4.4      2020-04-09 [1] CRAN (R 3.6.3)
#>  rlang           0.4.6      2020-05-02 [1] CRAN (R 3.6.3)
#>  rmarkdown       2.2        2020-05-31 [1] CRAN (R 3.6.3)
#>  rpart           4.1-15     2019-04-12 [2] CRAN (R 3.6.3)
#>  rprojroot       1.3-2      2018-01-03 [1] CRAN (R 3.6.1)
#>  rsample       * 0.0.7      2020-06-04 [1] CRAN (R 3.6.3)
#>  rsconnect       0.8.16     2019-12-13 [1] CRAN (R 3.6.3)
#>  rstan           2.19.3     2020-02-11 [1] CRAN (R 3.6.3)
#>  rstanarm        2.19.3     2020-02-11 [1] CRAN (R 3.6.3)
#>  rstantools      2.0.0      2019-09-15 [1] CRAN (R 3.6.3)
#>  rstudioapi      0.11       2020-02-07 [1] CRAN (R 3.6.3)
#>  scales        * 1.1.1      2020-05-11 [1] CRAN (R 3.6.3)
#>  sessioninfo     1.1.1      2018-11-05 [1] CRAN (R 3.6.1)
#>  shiny           1.4.0.2    2020-03-13 [1] CRAN (R 3.6.3)
#>  shinyjs         1.1        2020-01-13 [1] CRAN (R 3.6.2)
#>  shinystan       2.5.0      2018-05-01 [1] CRAN (R 3.6.3)
#>  shinythemes     1.1.2      2018-11-06 [1] CRAN (R 3.6.3)
#>  SnowballC       0.7.0      2020-04-01 [1] CRAN (R 3.6.3)
#>  StanHeaders     2.21.0-5   2020-06-09 [1] CRAN (R 3.6.3)
#>  statmod         1.4.34     2020-02-17 [1] CRAN (R 3.6.3)
#>  stringi         1.4.6      2020-02-17 [1] CRAN (R 3.6.2)
#>  stringr         1.4.0      2019-02-10 [1] CRAN (R 3.6.1)
#>  survival        3.1-11     2020-03-07 [2] CRAN (R 3.6.3)
#>  testthat        2.3.2      2020-03-02 [1] CRAN (R 3.6.3)
#>  threejs         0.3.3      2020-01-21 [1] CRAN (R 3.6.3)
#>  tibble        * 3.0.1      2020-04-20 [1] CRAN (R 3.6.3)
#>  tidymodels    * 0.1.0      2020-02-16 [1] CRAN (R 3.6.3)
#>  tidyposterior   0.0.3      2020-06-11 [1] CRAN (R 3.6.3)
#>  tidypredict     0.4.5      2020-02-10 [1] CRAN (R 3.6.3)
#>  tidyr           1.1.0      2020-05-20 [1] CRAN (R 3.6.3)
#>  tidyselect      1.1.0      2020-05-11 [1] CRAN (R 3.6.3)
#>  tidytext        0.2.4      2020-04-17 [1] CRAN (R 3.6.3)
#>  timeDate        3043.102   2018-02-21 [1] CRAN (R 3.6.0)
#>  tokenizers      0.2.1      2018-03-29 [1] CRAN (R 3.6.1)
#>  tune          * 0.1.0      2020-04-02 [1] CRAN (R 3.6.3)
#>  usethis         1.6.1      2020-04-29 [1] CRAN (R 3.6.3)
#>  vctrs           0.3.1      2020-06-05 [1] CRAN (R 3.6.3)
#>  withr           2.2.0      2020-04-20 [1] CRAN (R 3.6.3)
#>  workflows     * 0.1.1      2020-03-17 [1] CRAN (R 3.6.2)
#>  xfun            0.14       2020-05-20 [1] CRAN (R 3.6.3)
#>  xtable          1.8-4      2019-04-21 [1] CRAN (R 3.6.0)
#>  xts             0.12-0     2020-01-19 [1] CRAN (R 3.6.3)
#>  yaml            2.2.1      2020-02-01 [1] CRAN (R 3.6.1)
#>  yardstick     * 0.0.6      2020-03-17 [1] CRAN (R 3.6.2)
#>  zoo             1.8-8      2020-05-02 [1] CRAN (R 3.6.3)
#> 
#> [1] D:/R/Library/3.6
#> [2] C:/Program Files/R/R-3.6.3/library
```

**Comment by juliasilge** [comment]
The argument for `svm_rbf()` needs to be changed to `svm_margin` here and throughout:

https://github.com/tidymodels/parsnip/blob/7860b56e6808f6aae16f0a42f52f3ad7e6444201/R/svm_rbf_data.R#L35

**Comment by topepo** [comment]
Already done (ecf285b195a). I'll put in a PR in a bit.

**Comment by topepo** [comment]
See PR in #331 

``` r
library(tidymodels)
#> ── Attaching packages ───────────────────────── tidymodels 0.1.0.9000 ──
#> ✓ broom     0.5.6           ✓ recipes   0.1.12.9000
#> ✓ dials     0.0.7           ✓ rsample   0.0.7      
#> ✓ dplyr     1.0.0           ✓ tibble    3.0.1      
#> ✓ ggplot2   3.3.1           ✓ tune      0.1.0.9000 
#> ✓ infer     0.5.1           ✓ workflows 0.1.1      
#> ✓ parsnip   0.1.1.9000      ✓ yardstick 0.0.6      
#> ✓ purrr     0.3.4
#> ── Conflicts ───────────────────────────────── tidymodels_conflicts() ──
#> x purrr::discard() masks scales::discard()
#> x dplyr::filter()  masks stats::filter()
#> x dplyr::lag()     masks stats::lag()
#> x recipes::step()  masks stats::step()

get_from_env("svm_rbf_args") %>% 
  filter(parsnip == "margin") %>%
  pluck("func")
#> [[1]]
#> [[1]]$pkg
#> [1] "dials"
#> 
#> [[1]]$fun
#> [1] "svm_margin"

svm_rbf(margin = tune()) %>% 
  set_engine("kernlab") %>% 
  parameters() 
#> Collection of 1 parameters for tuning
#> 
#>      id parameter type object class
#>  margin         margin    nparam[+]

svm_rbf(margin = tune()) %>% 
  set_engine("kernlab") %>% 
  parameters() %>% 
  pull(object) %>% 
  str()
#> List of 1
#>  $ :List of 7
#>   ..$ type     : chr "double"
#>   ..$ range    :List of 2
#>   .. ..$ lower: num 0
#>   .. ..$ upper: num 0.2
#>   ..$ inclusive: Named logi [1:2] TRUE TRUE
#>   .. ..- attr(*, "names")= chr [1:2] "lower" "upper"
#>   ..$ trans    : NULL
#>   ..$ default  : language unknown()
#>   ..$ label    : Named chr "Insensitivity Margin"
#>   .. ..- attr(*, "names")= chr "margin"
#>   ..$ finalize : NULL
#>   ..- attr(*, "class")= chr [1:2] "quant_param" "param"
```

<sup>Created on 2020-06-16 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by topepo** [comment]
Fixed by #331
